__tests__/integration/integration.test.ts

```diff
@@ -0,0 +1,232 @@
+            Properties: {
+              'Octopus.Action.RunOnServer': 'false',
+              'Octopus.Action.Script.ScriptSource': 'Inline',
+              'Octopus.Action.Script.Syntax': 'Bash',
+              'Octopus.Action.Script.ScriptBody': "ehco 'hello'"
+            },
+            Links: {}
+          }
+        ]
+      }
+    ]
+    await repository.deploymentProcesses.saveToProject(
+      project,
+      deploymentProcess
+    )
+
+    const runbook = await repository.runbooks.create({
+      ProjectId: project.Id,
+      Description: 'Test Run book',
+      DefaultGuidedFailureMode: GuidedFailureMode.EnvironmentDefault,
+      Name: 'TestRunbook',
+      EnvironmentScope: RunbookEnvironmentScope.All,
+      MultiTenancyMode: TenantedDeploymentMode.TenantedOrUntenanted,
+      RunRetentionPolicy: {
+        QuantityToKeep: 1,
+        ShouldKeepForever: false
+      }
+    })
+    globalCleanup.add(async () => repository.runbooks.del(runbook))
+    const runbookProcess = await repository.runbookProcess.get(
+      runbook.RunbookProcessId,
+      undefined
+    )
+    runbookProcess.Steps = [
+      {
+        Condition: RunCondition.Success,
+        Links: {},
+        PackageRequirement: PackageRequirement.LetOctopusDecide,
+        StartTrigger: StartTrigger.StartAfterPrevious,
+        Id: '',
+        Name: `Run a Script`,
+        Properties: {'Octopus.Action.TargetRoles': 'deploy'},
+        Actions: [
+          {
+            Id: '',
+            Name: 'Run a Script',
+            ActionType: 'Octopus.Script',
+            Notes: null,
+            IsDisabled: false,
+            WorkerPoolId: null,
+            Container: {
+              Image: null,
+              FeedId: null
+            },
+            WorkerPoolVariable: null,
+            Environments: [],
+            ExcludedEnvironments: [],
+            Channels: [],
+            TenantTags: [],
+            CanBeUsedForProjectVersioning: false,
+            Packages: [],
+            Properties: {
+              'Octopus.Action.RunOnServer': 'false',
+              'Octopus.Action.Script.ScriptSource': 'Inline',
+              'Octopus.Action.Script.Syntax': 'Bash',
+              'Octopus.Action.Script.ScriptBody': "ehco 'hello'"
+            },
+            Links: {},
+            IsRequired: false
+          }
+        ]
+      }
+    ]
+    await repository.runbookProcess.save(runbookProcess)
+    let snapshot = {} as RunbookSnapshotResource
+    snapshot.ProjectId = project.Id
+    snapshot.Name = `Snapshot${runId}`
+    snapshot.RunbookId = runbook.Id
+    snapshot = await repository.runbookSnapshots.create(snapshot, {
+      publish: true
+    })
+    standardInput.runbook = runbook.Id
+    const env = await repository.environments.create({
+      Name: `Test-${runId}`
+    })
+    globalCleanup.add(async () => repository.environments.del(env))
+    globalCleanup.add(async () => new Promise(r => setTimeout(r, 2500)))
```

Comment:
? why is there a 2.5 second sleep here? Also async lambda shouldn't be neccessary